cli/manageDeployment.sh

```diff
@@ -202,7 +208,7 @@ function process_deployment() {
         if [[ "${DEPLOYMENT_SCOPE}" == "resourceGroup" ]]; then
 
           # Check resource group status
-          info "Creating resource group ${RESOURCE_GROUP} if required..."
+          info "${DRYRUN}Creating resource group ${RESOURCE_GROUP} if required..."
```

Comment:
Do you need to have the resource groups registered to run the plan? might unintentionally make changes that aren't expected from a read only change

Reply:
Yeah unfortunately you do. You can't even validate a template without an existing resource group. Demonstrations below.

As a work around for my own development I typically just use a resource group called "validator" - I could work that into the script instead? I could see about locking it down too so it can't be deployed to.




<details>
<summary>Dry Run as-is</summary>
<br>

```sh
master└─ $ $GENERATION_DIR/manageDeployment.sh -l application -u containerhost -g flibbidy -y
(Info) (Dryrun) Preparing the context...
(Info) (Dryrun) processing the deployment ...
(Info) (Dryrun) Registering Resource Providers.
 [ ✔ ]  Microsoft.Web
(Info) (Dryrun) Creating resource group flibbidy if required...
(Info) Validating template...
(Info) Template is valid.
(Info) (Dryrun) Starting deployment of resourceGroup-application-containerhost to the Resource Group flibbidy.
This command is in preview. It may be changed/removed in a future release.
(Info) (Dryrun) Results for resourceGroup-application-containerhost
{
  "changes": [
    {
      "after": {
        "apiVersion": "2020-06-01",
        "id": "/subscriptions/02b9b749-b541-4cc3-9ae2-6f9d4a4d9533/resourceGroups/flibbidy/providers/Microsoft.Web/serverfarms/djankywalks-integration-rossmurr4y-appli",
        "kind": "linux",
        "location": "eastus",
        "name": "djankywalks-integration-rossmurr4y-appli",
        "properties": {
          "reserved": true
        },
        "resourceGroup": "flibbidy",
        "sku": {
          "capacity": 1,
          "name": "P1v2"
        },
        "type": "Microsoft.Web/serverfarms"
      },
      "before": null,
      "changeType": "Create",
      "delta": null,
      "resourceId": "/subscriptions/02b9b749-b541-4cc3-9ae2-6f9d4a4d9533/resourceGroups/flibbidy/providers/Microsoft.Web/serverfarms/djankywalks-integration-rossmurr4y-appli"
    }
  ],
  "error": null,
  "status": "Succeeded"
}
(Info) (Dryrun) update completed for flibbidy.
```

</details>


<details>
<summary>Dry-run without resource group creation, but including attempted validation</summary>
<br>

```sh
hamlet @ ~/cmdb/cmdb/gs/msw-cmdb/djankywalks/config/solutionsv2/integration/rossmurr4y
master└─ $ $GENERATION_DIR/manageDeployment.sh -l application -u containerhost -g flibbidydibbidy -y
(Info) (Dryrun) Preparing the context...
(Info) (Dryrun) processing the deployment ...
(Info) (Dryrun) Registering Resource Providers.
 [ ✔ ]  Microsoft.Web
(Info) Validating template...
Azure Error: ResourceGroupNotFound
Message: Resource group 'flibbidydibbidy' could not be found.
(Fatal) There was an issue during deployment.
```

</details>

<details>
<summary>Dry run with neither resource group creation or validation</summary>
<br>

```sh
hamlet @ ~/cmdb/cmdb/gs/msw-cmdb/djankywalks/config/solutionsv2/integration/rossmurr4y
master└─ $ $GENERATION_DIR/manageDeployment.sh -l application -u containerhost -g flibbidydibbidy -y
(Info) (Dryrun) Preparing the context...
(Info) (Dryrun) processing the deployment ...
(Info) (Dryrun) Registering Resource Providers.
 [ ✔ ]  Microsoft.Web
(Info) (Dryrun) Starting deployment of resourceGroup-application-containerhost to the Resource Group flibbidydibbidy.
This command is in preview. It may be changed/removed in a future release.
Azure Error: ResourceGroupNotFound
Message: Resource group 'flibbidydibbidy' could not be found.
(Fatal) There was an issue during deployment.
```

</details>